x/liquidity/spec/01_concepts.md

```diff
@@ -4,88 +4,60 @@
 
-- Combines a traditional order book-based exchange system with a Uniswap-like AMM mechanism. This hybrid system deepens liquidity for the token swap marketplace.
+The ESPM model is intended to provide protection against price volatility, transaction ordering vulnerabilities, and losses due to arbitrage. AMMs such as Uniswap do not provide protection against these whatever they are.
 
-- Executes batch-style swaps that minimize front-running risk and sub-second latency competition, thereby protecting ordinary traders.
+## Batch Execution
 
-  - The order book accumulates incoming limit orders into a batch.
-  - The liquidity module matches accumulated limit orders and orders from the liquidity pool at an equivalent swap price at each batch execution height.
-  - All limit orders in a batch are treated equally and executed at the same swap price.
+The liquidity module uses a batch execution methodology. Deposits, withdrawals, and swap orders are accumulated in a liquidity pool for a pre-defined period that is one or more blocks in length. Orders are then added to the pool and executed at the end of the batch. The size of each batch is configured by using the `UnitBatchSize` governance parameter.
 
-## Features of the Liquidity Module
+## Price Discovery
 
-The main features of the liquidity module are:
+Swap prices in liquidity pools are determined by the current pool coin reserves and the requested swap amount. Arbitrageurs buy or sell coins in liquidity pools to gain instant profit that results in real-time price discovery of liquidity pools.
 
-### Liquidity Pool
+## Escrow Process
 
-A liquidity pool is a coin reserve that contains two kinds of coins in a trading pair. A unique pool exists for each token pair.
+The liquidity module uses a module account that acts as an escrow account. The module account holds and releases the coin amount during batch execution.
 
-A liquidity provider (entity or person) deposits the coins into the liquidity pool and then shares the accumulated swap fees with respect to their pool share. Pool share is represented as possession of pool coins.
+## Refund 
 
-A liquidity pool is permissionless.
+The liquidity module has refund functions when deposit, withdraw, or swap batch states are not successfully executed.
+Read [the logic](https://github.com/tendermint/liquidity/blob/e8ab2f4d75079157d008eba9f310b199573eed28/x/liquidity/keeper/batch.go#L83-L127) to get more context.
+## Fees
+### PoolCreationFee
 
-### Coin Swap
+The liquidity module `PoolCreationFee` parameter is paid upon pool creation. This param is defined at genesis and can be updated by using the governance proposal. The purpose of this fee is to prevent users from creating useless pools and making limitless transactions. The funds from this fee go to the community fund.
```

Comment:
```suggestion
The liquidity module pool creation fee set by the `PoolCreationFee` parameter is paid on pool creation. This parameter is defined at genesis and can be updated by using the governance proposal. The purpose of this fee is to prevent users from creating useless pools and making limitless transactions. The funds from this fee go to the community fund.
```